esp/services/ws_workunits/ws_workunitsQuerySets.cpp

```diff
@@ -2740,3 +2741,68 @@ bool CWsWorkunitsEx::onWUUpdateQueryEntry(IEspContext& context, IEspWUUpdateQuer
     }
     return true;
 }
+
+bool CWsWorkunitsEx::onWUGetNumFileToCopy(IEspContext& context, IEspWUGetNumFileToCopyRequest& req, IEspWUGetNumFileToCopyResponse& resp)
+{
+    try
+    {
+        Owned<IPropertyTree> result;
+        StringBuffer clusterName = req.getClusterName();
+        if (!clusterName.isEmpty())
+        {
+            SocketEndpointArray servers;
+            getRoxieProcessServers(clusterName.str(), servers);
+            if (servers.length() < 1)
+                throw MakeStringException(ECLWATCH_CANNOT_RESOLVE_CLUSTER_NAME, "Process Server not found for %s", clusterName.str());
+            result.setown(sendRoxieControlAllNodes(servers.item(0), "<control:numfilestoprocess/>", false, ROXIELOCKCONNECTIONTIMEOUT));
+        }
+        else
+        {
+            StringBuffer url = req.getClusterURL();
```

Comment:
I don't think you need to support getting the info by IP, unless Miguel asked for it?  It's a bit of a security hole as it can be used to send a control message anywhere.